### PR TITLE
Scripts/Magisters Terrace: move NPC Pure Energy to SAI

### PR DESCRIPTION
**Pull request by Gooyeth**
Change Script to SAI and crash fix:

```c++
Unit::CastSpell (this=0x0, target=0x0, spellId=44335, args=...)
npc_pure_energy::npc_pure_energyAI::JustDied (this=0x7ffe812cdf10, killer=0x0)
```

**Target branch(es):** 3.3.5/master

- [X] 3.3.5
- [x] master

**Tests performed:** build, tested in-game

### CODE REVIEW

**Comment by jackpoz** [comment]
Did you miss the sai script ?

**Comment by Keader** [comment]
He did here Jack: https://github.com/TrinityCore/TrinityCore/issues/21357

**Comment by jackpoz** [comment]
well the SQL script has to be in this PR

**Comment by Gooyeth** [comment]
Ready and tested in-game

File: sql/updates/world/master/2018_02_04_21358.sql [hunk 1]
```diff
@@ -3,6 +3,6 @@ SET @ENTRY := 24745;
 UPDATE `creature_template` SET `AIName`="SmartAI", `ScriptName` = "" WHERE `entry`=@ENTRY;
 DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
 INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
-(@ENTRY,0,0,0,6,0,100,0,0,0,0,0,11,44335,2,0,0,0,0,7,0,0,0,0,0,0,0,"Pure Energy - On Just Died - Cast 'Energy Feedback'"),
+(@ENTRY,0,0,0,6,0,100,0,0,0,0,0,75,44335,0,0,0,0,0,7,0,0,0,0,0,0,0,"Pure Energy - On Just Died - Add Aura 'Energy Feedback'"),
```
Comment (Killyana) [inline sql/updates/world/master/2018_02_04_21358.sql]: why this change?

Comment (Gooyeth) [reply sql/updates/world/master/2018_02_04_21358.sql]: Originally:
```c++
killer->CastSpell(killer...);
```
However, SMART_ACTION_CAST does not work on SMART_EVENT_DEATH, at least not with this spell.

Comment (Killyana) [reply sql/updates/world/master/2018_02_04_21358.sql]: It must work with the cast flag=2

Comment (Gooyeth) [reply sql/updates/world/master/2018_02_04_21358.sql]: It does not work, it's because the target is TARGET_UNIT_CASTER in spell.

**Comment by Killyana** [comment]
The npc has https://woehead.way-of-elendil.fr/?spell=44326 on creature_template_addon that trigger https://woehead.way-of-elendil.fr/?spell=36294 it's used to set a react passive, but it doesn't work, probably it must be fixed on spell_dbc

**Comment by Killyana** [comment]
This spell is also used http://www.wowhead.com/spell=44328/energy-feedback it's casted in a random target but it mustn't be interrupted by the movement, and it doesn't deal any damage.

**Comment by Gooyeth** [comment]
Follow a player and do not use melee, I'll wait for the sniffer.

**Comment by Gooyeth** [comment]
Talking to Shauren, he told me he will not update the sniffer anymore

**Comment by Gooyeth** [comment]
I have the sniffer.

**Comment by kvipka** [comment]
i think it's wrong way to convert boss scripts in SAI. For trash or usual npc okay, but not for boss

**Comment by Keader** [comment]
I agree with tou kvipka

File: src/server/scripts/EasternKingdoms/MagistersTerrace/boss_vexallus.cpp [hunk 2]
```diff
@@ -187,40 +187,7 @@ class boss_vexallus : public CreatureScript
...
-
-        struct npc_pure_energyAI : public ScriptedAI
-        {
-            npc_pure_energyAI(Creature* creature) : ScriptedAI(creature)
-            {
-                me->SetDisplayId(me->GetCreatureTemplate()->Modelid2);
-            }
-
-            void JustDied(Unit* killer) override
-            {
```
Comment (kvipka) [inline src/server/scripts/EasternKingdoms/MagistersTerrace/boss_vexallus.cpp]: just add check `if (killer)`

Comment (Keader) [reply src/server/scripts/EasternKingdoms/MagistersTerrace/boss_vexallus.cpp]: Yes, i will fix it when i back of uni today.

**Comment by Keader** [comment]
If has some improvement in boss script, please open a new PR.
Crash fixed in 5bef3e426aea37b0e7555f146037c58e7be6352d

**Comment by Gooyeth** [comment]
#21356 With pleasure 😅
